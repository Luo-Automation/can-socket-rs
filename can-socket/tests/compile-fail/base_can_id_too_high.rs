fn main() {
	let id = can_socket::base_id!(0x800);
	println!("{id:?}");
}
