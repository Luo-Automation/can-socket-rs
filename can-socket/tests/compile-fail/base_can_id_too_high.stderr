error[E0080]: evaluation of `main::{constant#0}` failed
 --> tests/compile-fail/base_can_id_too_high.rs:2:11
  |
2 |     let id = can_socket::base_id!(0x800);
  |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^ the evaluated program panicked at 'invalid base CAN ID', $DIR/tests/compile-fail/base_can_id_too_high.rs:2:14
  |
  = note: this error originates in the macro `$crate::panic::panic_2021` which comes from the expansion of the macro `can_socket::base_id` (in Nightly builds, run with -Z macro-backtrace for more info)

note: erroneous constant encountered
 --> tests/compile-fail/base_can_id_too_high.rs:2:11
  |
2 |     let id = can_socket::base_id!(0x800);
  |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this note originates in the macro `can_socket::base_id` (in Nightly builds, run with -Z macro-backtrace for more info)
